sql/src/main/java/io/crate/operation/merge/IteratorPageDownstream.java

```diff
@@ -117,6 +117,10 @@ private void processBuckets(Iterator<Row> iterator, PageConsumeListener listener
                 return;
             }
         }
+        if (topRowUpstream.shouldPause()) {
```

Comment:
why is this required here? This is already done directly after the setNextRow call?